File: dagger-mvp/src/main/java/be/vergauwen/simon/daggermvp/ui/TestContract.java
package be.vergauwen.simon.daggermvp.ui;

import be.vergauwen.simon.himura.MVPContract;

public interface TestContract {
    interface View extends MVPContract.View {
        void showThings();
    }

    interface Presenter extends MVPContract.Presenter<View> {
        void loadThings();
    }

    interface Component extends MVPContract.Component<Presenter> {}
}

File: app/src/main/kotlin/be/vergauwen/simon/ui/TestContract.kt
package be.vergauwen.simon.ui

import be.vergauwen.simon.mvp.MVPContract

interface TestContract {
  interface View : MVPContract.View {
    fun showThings()
  }

  interface Presenter : MVPContract.Presenter<View> {
    fun loadThings()
  }
}